# Description	
Mapping of sample sequences against the reference genome is conducted with an aim of determining the most likey source of the observed sequencing reads.
`BWA-MEM` was used for alignment. The results of mapping is a sequence alignment map (SAM) format. 
The file has a single unified format for storing read alignments to a reference genome.

## Mapping Command
```
mkdir Mapping

# Download the human reference genome
wget -c ftp://ftp.ensembl.org/pub/release-100/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz -O Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz

gunzip Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz 
	
#Index reference file	
bwa index Homo_sapiens.GRCh38.dna.primary_assembly.fa
	
#Perform alignment
for sample in `cat sample_id.txt`
do 
bwa mem hg19.chr5_12_17.fa Trim_galore/${sample}_R1_val_1.fq.gz  \
       Trim_galore/${sample}_R2_val_2.fq.gz > Mapping/${sample}.sam
done
	
```
## Conversion of the SAM file to BAM file, sorting and indexing
A Binary Alignment Map (BAM) format is an equivalent to sam but its developed for fast processing and indexing. It stores every read base, base quality and uses a single conventional technique for all types of data.
The produced BAM files were sorted by read name and indexing was done for faster or rapid  retrieval. At the end of the every BAM file,  a special end of file (EOF) marker is usually written, the samtools index command also checks for this and produces an error message if its not found.
	
```
for sample in `cat sample_id.txt`
do
        Convert SAM to BAM and sort it 
        samtools view -@ 20 -S -b Mapping/${sample}.sam | samtools sort -@ 32 > Mapping/${sample}.sorted.bam
        
        Index BAM file
        samtools index Mapping/${sample}.sorted.bam
done
```

## Mapped reads filtering
### Option 1 <rmdup>
```
samtools rmdup sample1.sorted.bam  SLGFSK35.rdup and samtools rmdup sample2.sorted.bam  SLGFSK36.rdup.
```

 ### Option 2
```
for sample in `cat sample_id.txt`
do
	#Filter BAM files
        samtools view -q 1 -f 0x2 -F 0x8 -b Mapping/${sample}.sorted.bam > Mapping/${sample}.filtered1.bam
done
```

To view the output of the results use :
```
samtools flagstat <bam file>
```
#### Duplicates removal
During library construction sometimes there's introduction of PCR (Polymerase Chain Reaction) duplicates, these duplicates usually can result in false SNPs (Single Nucleotide Polymorphisms), whereby the can manifest themselves as high read depth support. A low number of duplicates (<5%) in good libraries is considered standard.

```
#use the command <markdup>
for sample in `cat sample_id.txt`
do
	samtools collate -o Mapping/${sample}.namecollate.bam Mapping/${sample}.filtered1.bam
        samtools fixmate -m Mapping/${sample}.namecollate.bam Mapping/${sample}.fixmate.bam
        samtools sort -@ 32 -o Mapping/${sample}.positionsort.bam Mapping/${sample}.fixmate.bam
        samtools markdup -@32 -r Mapping/${sample}.positionsort.bam Mapping/${sample}.clean.bam
done
```
